tls/s2n_handshake_io.c

```diff
@@ -875,6 +899,83 @@ static int s2n_try_delete_session_cache(struct s2n_connection *conn)
     return 0;
 }
 
+/* HelloRetryRequests need to recreate the transcript after the ClientHello2 message is received.
+ * Doing this requires a synthetic message header, a hash of the ClientHello1 message, and a
+ * hash of the HelloRetryRequest. This should be done without having to store any state. To achieve
+ * this, here is where we get the data:
+ *
+ * ClientHello1 => This is a hash value that is stored in the cookie extension.
+ * HelloRetryRequest => This message should be re-created based on the ClientHello values.
+ *                      This works because nothing should change between ClientHello1 and
+ *                      ClientHello2 aside from the key share.
+ * ClientHello2 => This is the message currently in conn->handshake.io. We don't need to
```

Comment:
why we are recreating the hrr? It's not safe to assume that nothing would change between CH1 and CH2

Reply:
No longer using cookies to store the hash. This simplifies recreating the hash transcript.